uast/uast-kotlin-base/src/org/jetbrains/uast/kotlin/psi/UastKotlinPsiVariable.kt

```diff
@@ -77,76 +76,81 @@ class UastKotlinPsiVariable private constructor(
-                    containingElement = containingElement,
-                    ktElement = initializer
-                )
+            UastKotlinPsiVariable(
+                manager = initializer.manager,
+                name = "var" + Integer.toHexString(initializer.getHashCode()),
+                typeProducer = {
+                    val service = ServiceManager.getService(BaseKotlinUastResolveProviderService::class.java)
+                    service.getType(initializer, containingElement) ?: UastErrorType
+               },
+                ktInitializer = initializer,
+                psiParentProducer = { containingElement.getParentOfType<UDeclaration>()?.psi ?: parent },
+                containingElement = containingElement,
+                ktElement = initializer
+            )
 
         fun create(
-            baseKotlinUastResolveProviderService: BaseKotlinUastResolveProviderService,
             name: String,
             localFunction: KtFunction,
             containingElement: UElement
         ): PsiLocalVariable =
-                UastKotlinPsiVariable(
-                    baseKotlinUastResolveProviderService,
-                    manager = localFunction.manager,
-                    name = name,
-                    typeProducer = {
-                        baseKotlinUastResolveProviderService.getFunctionType(localFunction, containingElement) ?: UastErrorType
-                    },
-                    ktInitializer = localFunction,
-                    psiParentProducer = { containingElement.getParentOfType<UDeclaration>()?.psi ?: localFunction.parent },
-                    containingElement = containingElement,
-                    ktElement = localFunction
-                )
+            UastKotlinPsiVariable(
+                manager = localFunction.manager,
+                name = name,
+                typeProducer = {
+                    val service = ServiceManager.getService(BaseKotlinUastResolveProviderService::class.java)
+                    service.getFunctionType(localFunction, containingElement) ?: UastErrorType
+                },
+                ktInitializer = localFunction,
+                psiParentProducer = { containingElement.getParentOfType<UDeclaration>()?.psi ?: localFunction.parent },
+                containingElement = containingElement,
+                ktElement = localFunction
+            )
     }
 }
 
 private class KotlinUastPsiExpression(
-    private val baseKotlinUastResolveProviderService: BaseKotlinUastResolveProviderService,
     val ktExpression: KtExpression,
     val parent: UElement
 ) : PsiElement by ktExpression, PsiExpression {
-    override fun getType(): PsiType? = baseKotlinUastResolveProviderService.getType(ktExpression, parent)
+    override fun getType(): PsiType? {
+        val service = ServiceManager.getService(BaseKotlinUastResolveProviderService::class.java)
```

Comment:
Btw, there is `com.intellij.openapi.components.service` helper available. So, you may write  `service<BaseKotlinUastResolveProviderService>()` instead

Reply:
Thanks for the suggestion, but adding that in IJ doesn't find anything. `com.intellij.openapi.components` points to https://github.com/JetBrains/intellij-community/tree/master/platform/core-api/src/com/intellij/openapi/components and I'm not sure there is such util.